File: sic-api/Entities/Su/SuUpload.cs
using Microsoft.EntityFrameworkCore;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using sic_api.Model.Storage;

namespace sic_api.Entities.Su;

[Index(nameof(BucketName))]
[Index(nameof(ObjectKey))]
[Table("su_upload")]
public class SuUpload : BaseEntity
{
    [Column("bussiness_id")]
    public Guid? BusinessId { get; set; }

    [Required]
    [MaxLength(100)]
    [Column("bucket_name")]
    public string BucketName { get; set; } = default!;

    [Required]
    [MaxLength(1000)]
    [Column("object_key")]
    public string ObjectKey { get; set; } = default!;

    [Required]
    [MaxLength(255)]
    [Column("file_name")]
    public string FileName { get; set; } = default!;

    [Required]
    [MaxLength(255)]
    [Column("content_type")]
    public string ContentType { get; set; } = default!;

    [Required]
    [Column("file_size")]
    public long FileSize { get; set; } = 0;

    [Required]
    [MaxLength(50)]
    [Column("category")]
    public string Category { get; set; } = default!;

    [Required]
    [Column("visibility")]
    public FileVisibility Visibility { get; set; }

    [Required]
    [MaxLength(2000)]
    [Column("storage_url")]
    public string StorageUrl { get; set; } = default!;

    [Required]
    [MaxLength(2000)]
    [Column("access_url")]
    public string AccessUrl { get; set; } = default!;

    [Required]
    [Column("is_streaming")]
    public bool IsStreaming { get; set; } = false;

    [Required]
    [Column("is_active")]
    public bool IsActive { get; set; } = false;

    [Column("upload_group_id")]
    public Guid? UploadGroupId { get; set; } = null;

    [Column("temp_expires_at")]
    public DateTime? TempExpiresAt { get; set; }
}

File: sic-spring/sic/src/main/java/com/softinter/sicapi/entity/su/SuUpload.java
package com.softinter.sicapi.entity.su;

import com.softinter.sicapi.entity.base.BaseEntity;
import com.softinter.sicapi.entity.enums.FileCategory;
import com.softinter.sicapi.entity.enums.FileVisibility;

import jakarta.persistence.*;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

import java.time.Instant;
import java.util.UUID;

@Entity
@Table(name = "su_upload",
       indexes = {
           @Index(name = "idx_bucket_name", columnList = "bucket_name"),
           @Index(name = "idx_object_key", columnList = "object_key")
       })
@Data
@EqualsAndHashCode(callSuper = true)
@NoArgsConstructor
public class SuUpload extends BaseEntity {

    @Column(name = "bussiness_id") // สะกดตาม C# (bussiness_id)
    private UUID businessId;

    @Column(name = "bucket_name", nullable = false, length = 100)
    private String bucketName;

    @Column(name = "object_key", nullable = false, length = 1000)
    private String objectKey;

    @Column(name = "file_name", nullable = false, length = 255)
    private String fileName;

    @Column(name = "content_type", nullable = false, length = 255)
    private String contentType;

    @Column(name = "file_size", nullable = false)
    private Long fileSize = 0L;

    @Column(name = "category", nullable = false, length = 50)
    private String category;

    @Enumerated(EnumType.ORDINAL)
    @Column(name = "visibility", nullable = false)
    private FileVisibility visibility;

    @Column(name = "storage_url", nullable = false, length = 2000)
    private String storageUrl;

    @Column(name = "access_url", nullable = false, length = 2000)
    private String accessUrl;

    @Column(name = "is_streaming", nullable = false)
    private Boolean isStreaming = false;

    @Column(name = "is_active", nullable = false)
    private Boolean isActive = false;

    @Column(name = "upload_group_id")
    private UUID uploadGroupId;

    @Column(name = "temp_expires_at")
    private Instant tempExpiresAt;

    @Column(name = "mime_type", nullable = false)
    private String mimeType;

    @Enumerated(EnumType.STRING)
    @Column(name = "file_category", nullable = false)
    private FileCategory fileCategory;

    @Enumerated(EnumType.STRING)
    @Column(name = "file_visibility", nullable = false)
    private FileVisibility fileVisibility;

  
}
